src/main/kotlin/uk/gov/dwp/dataworks/aws/AwsCommunicator.kt

```diff
@@ -449,6 +464,92 @@ class AwsCommunicator {
         }
     }
 
+    /**
+     * Retrieves role details for the user [userName]. If the role does not exist
+     * this method returns null.
+     */
+    fun getIamRole(correlationId: String, userName: String): Role? {
+        val roleName = "orchestration-service-user-$userName-role"
+
+        return try {
+            val role = awsClients.iamClient.getRole(GetRoleRequest.builder()
+                .roleName(roleName)
+                .build()).role()
+            logger.info("Role already exists for $userName",
+                "correlation_id" to correlationId,
+                "role_arn" to role.arn(),
+                "role_path" to role.path(),
+                "role_name" to role.roleName(),
+                "role_id" to role.roleId(),
+                "created_date" to role.createDate().toString())
+            updateDynamoDeploymentEntry(userName, "iamRoleName" to role.roleName())
+            role
+        } catch (e: IamException) {
+            logger.info("Role does not exist for $userName",
+                "correlation_id" to correlationId,
+                        "user" to userName,
+                        "exception" to e.localizedMessage)
+            null
+        }
+
+    }
+
+    fun listRolesWithPrefix(pathPrefix: String): ListRolesResponse{
+        return awsClients.iamClient.listRoles(
```

Comment:
Check this API call isnt limited to 10 results!

Reply:
[The documentation ](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/iam/IamClient.html#listRoles-software.amazon.awssdk.services.iam.model.ListRolesRequest-) mentions that you _can_ paginate the results if you specify `MaxItems`, so it shouldn't happen by default. Either way, the roles would get deleted the next day.